webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -138,40 +134,56 @@ class TechniqueWriter (
       cr1.copy(directives = cr1.directives ++ cr2.directives)
     }
 
-    for {
-      directives <- readDirectives.getFullDirectiveLibrary().map(_.allActiveTechniques.values.filter(_.techniqueName.value == techniqueId.name.value).flatMap(_.directives).filter(_.techniqueVersion == techniqueId.version))
-      technique  <- techniqueRepository.get(techniqueId).notOptional(s"No Technique with ID '${techniqueId.toString()}' found in reference library.")
```

Comment:
You should extract the for loop in a function (maybe an inner function, it does not need to be global for now)

Same for the none part, it should be in an inner function